### PR TITLE
Refactor dom-if & dom-repeat using Polymer.Element.

### PR DESCRIPTION
**Pull request by kevinpschaaf**
Refactor dom-if & dom-repeat using Polymer.Element.  Eliminates dependency on legacy Polymer API and related bulk.

### CODE REVIEW

File: src/templatizer/dom-if.html [hunk 1]
```diff
@@ -30,63 +30,78 @@
    * Set the `restamp` property to true to force the stamped content to be
    * created / destroyed when the `if` condition changes.
    */
-  Polymer({
-
-    is: 'dom-if',
+  class DomIf extends Polymer.Element {
+
+    static get is() { return 'dom-if' }
```
Comment (sorvell) [inline src/templatizer/dom-if.html]: You didn't use this when defining and there's no template. We should be able to omit it here then right?

File: src/templatizer/dom-repeat.html [hunk 2]
```diff
@@ -108,170 +108,179 @@
 
   var templatizer = new Polymer.Templatizer();
 
-  Polymer({
+  class DomRepeat extends Polymer.Element {
 
-    is: 'dom-repeat',
+    static get is() { return 'dom-repeat' }
```
Comment (sorvell) [inline src/templatizer/dom-repeat.html]: Same point as `dom-if`, can we remove this?

File: src/templatizer/dom-repeat.html [hunk 3]
```diff
@@ -397,31 +406,29 @@
...
           }
         }
       }
-    },
+    }
 
-    _debounceRender: function() {
-      Polymer.Templatizer.enqueueDebouncer(this.debounce('_render', this._render));
-    },
+    _debounceRender(fn, delay) {
```
Comment (sorvell) [inline src/templatizer/dom-repeat.html]: In `dom-if` a similar function is called `queueRender`. Any reason to name this differently?

File: src/templatizer/dom-repeat.html (hunk 2 above)
Comment (kevinpschaaf) [inline src/templatizer/dom-repeat.html]: Done, removed both.

File: src/templatizer/dom-repeat.html (hunk 3 above)
Comment (kevinpschaaf) [inline src/templatizer/dom-repeat.html]: Done, renamed `dom-if`'s `_queueRender` -> `_debounceRender`

**Comment by kevinpschaaf** [comment]
Rebasing against `2.0-preview`, which should fix test issues.

**Pull request merged**